EIPS/eip-restricted_chain_context_txs.md

```diff
@@ -0,0 +1,180 @@
+(the transactor) an ability to express requirements about the transaction's relationship
+to blockchain data and its provenance.
+
+- Introduces an opportunity for miners to compete for consumers' transactions (consumers can prefer miners, where -- under the status quo -- the current miner-transaction processing service is almost perfectly homogenous from the consumer perspective).
+
+## Specification
+
+
+### Parameters
+
+- `FORK_BLOCK_NUMBER` `TBD`
+- `TRANSACTION_TYPE_NUMBER` `0x45/TBD`.  See EIP-2718.
+
+As of `FORK_BLOCK_NUMBER`, a new EIP-2718 transaction is introduced with `TransactionType` `TRANSACTION_TYPE_NUMBER`.
+
+The EIP-2718 `TransactionPayload` for this transaction is `rlp([chainId, chainContext, nonce, gasPrice, gasLimit, to, value, data, access_list, yParity, senderR, senderS])`.
+
+
+### Definitions
+
+- `chainContext`. The transaction is only valid for blockchain data satisfying ALL OF the annotations.
+- `ANNOTATION_PREFIX`. An integer used to describe the set of subclass annotations in the `chainContext` (_ie._ _which_ chain context subclasses should the provided values be applied to).
+
+### Subclasses
+
+Annotation prefixes (`ANNOTATION_PREFIX`) are used to represent combinations of the available context subclasses.
+
+`chainContext` is of form `<ANNOTATION_PREFIX>[subclass...]`, where `...` means "zero or more of the things to the left." 
+
+Subclasses are defined independently, and can be modified independently from this proposal.
+
+#### `segmentId`
+
+- `ANNOTATION_PREFIX` `1`.
+- `SEGMENT_ID` `bytes`. A byte array between 4 and 12 bytes in length. 
+
+Constrains the validity of a transaction by a chain context subclass referencing prior canonical block by number and hash.
+The transaction is only valid when included in a block having the annotated block as an ancestor.
+
+It accomplishes this by describing a unique value referencing a canonical block, where the referenced block must be included in any chain segment also including this transaction.
+
+Practically, the "designated chain segment" can be understood as the segment of blocks from `0..segmentId` inclusive.
+
+This pattern can be understood as a correlate of EIP-155's `chainId` specification. EIP155 defines the restriction of transactions between chains; limiting the applicability of any EIP-155 transaction to a chain with the annotated ChainID. 
+`segmentId` further restricts transaction application to one subsection ("segment") of one chain.
+
+From this constraint hierarchy, we note that it is possible an implementation of `segmentId` to make `chainId` redundant.
+
+##### Construction
+
+The value is generally constructed as a concatenated `BlockNumber``BlockHashPrefix`, where `BlockHashPrefix` is __the first 4 bytes of the block hash of the canonical block with number `BlockNumber`__.
+
+If the referenced `BlockNumber` is `0` it may be omitted. Otherwise, `BlockNumber` should be a __little endian encoded value with trailing zero values truncated__.
```

Comment:
You don't need to do little endian encoding in order to benefit from the space saving, and the rest of the protocol uses big endian I believe.  Because of this, I recommend big endian.